File: 0392-is-subsequence/0392-is-subsequence.cs
public class Solution {
    public bool IsSubsequence(string s, string t) {
        int sidx = 0, tidx = 0;
        while(sidx < s.Length && tidx < t.Length){
            if(s[sidx] == t[tidx])
                sidx++;
            tidx++;
        }
        if(sidx == s.Length) return true;
        return false;
    }
}

File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int sidx = 0, tidx = 0;
        char[] sch = s.toCharArray(), tch = t.toCharArray();
        while(sidx < sch.length && tidx < tch.length){
            if(sch[sidx] == tch[tidx])
                sidx++;
            tidx++;
        }
        if(sidx == s.length()) return true;
        return false;
    }
}